### PR TITLE
support models from www.modelscope.cn

### PR DESCRIPTION
**Pull request by liuyhwangyh**
Motivation
Support models from [www.modelscope.cn](http://www.modelscope.cn/), running server.

### CODE REVIEW

File: python/sglang/srt/server.py [hunk 1]
```diff
@@ -250,6 +251,10 @@ def launch_server(
     )
     logger.info(f"{server_args=}")
 
+    # Use model from www.modelscope.cn, first download the model.
+    server_args.model_path = prepare_model(server_args.model_path)
+    server_args.tokenizer_path = prepare_model(server_args.tokenizer_path)
```
Comment (zhyncs) [inline python/sglang/srt/server.py]: For tokenizer, we should not download weights.

File: python/sglang/srt/utils.py [hunk 2]
```diff
@@ -701,3 +701,12 @@ async def authentication(request, call_next):
         if request.headers.get("Authorization") != "Bearer " + api_key:
             return JSONResponse(content={"error": "Unauthorized"}, status_code=401)
         return await call_next(request)
+
+
+def prepare_model(model_path):
```
Comment (zhyncs) [inline python/sglang/srt/utils.py:706]: Please add unit test for this.

**Review by zhyncs — APPROVED** [review]
@liuyhwangyh LGTM I will verify it locally, and if there are no issues, I'll merge it. Thanks for your contribution!

**Comment by zhyncs** [comment]
@liuyhwangyh May you fix the lint issue? Thanks. ref https://github.com/sgl-project/sglang/blob/main/docs/en/contributor_guide.md

File: python/sglang/srt/server.py (hunk 1 above)
Comment (liuyhwangyh) [reply python/sglang/srt/server.py]: ignore weight files.

File: python/sglang/srt/utils.py (hunk 2 above)
Comment (liuyhwangyh) [reply python/sglang/srt/utils.py:706]: done

**Pull request merged**